### ISSUE TITLE
Question: run InSpec test against each and every VM in a Subscription

### ISSUE THREAD

**Issue by matt-mcc1** [inspec/inspec-azure#176]
## Detailed Description
Hello

How can I create an InSpec test that would run a single test against _every_ VM in a Subscription? 

The example code for a VM test requires a VM resource group & name - how could I iterate through every VM in a subscription and ensure it passes InSpec assertions (e.g all VM locations should be WestEurope)?

Perhaps AZ CLI commands to then create a list of each VM name and location, then feed it into something like the below?

```
servers = attribute('server_list', default:[], description: 'Array of server info')

control 'Example' do

impact 1.0
title 'Integration test'

describe servers do
  it { should_not be_empty }
end
unless servers.empty?
  servers.each do |intern|
    compute = azure_virtual_machine(group_name: intern['resource_group_name'], name: intern['machine_name'])
     describe "intern['machine_name']" do
        subject {compute}
        its('location') { should cmp 'northeurope' }
        its('type') { should cmp 'Microsoft.Compute/virtualMachines' }
        its('name') { should cmp "intern['machine_name']" }
      end
    end
  end
end
```

Thanks in advance!

**Comment by rmoles** [comment]
Hi, @matt-mcc1 I was just wondering whether if this is still an issue? If not would you mind posting the resolution?

**Comment by omerdemirok** [comment]
@matt-mcc1, you can use the new resources, available since `v1.20.0`, to do that.

Please [here](https://github.com/inspec/inspec-azure/blob/master/docs/resources/azure_virtual_machines.md#loop-through-virtual-machines-by-their-ids) for an example.
